src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -1,3 +1,4 @@
+#if !NETSTANDARD2_0
```

Comment:
`Promote` is for going distributed. In the mono case, it seems likely to me it has to throw `NotSupportedException`. But for which use do you want to implement `IPromotableSinglePhaseNotification`? If it is in the "DtcFailures" tests, this could cause them to be no more distributed, which is not the aim of those tests. If it is in the `AdoNetWithDistributedTransactionFactory`, it should not be needed, and I do not believe it can be done while preserving the "flush on commit" feature. The 2 PC volatile ressource NHibernate is enlisting there should not cause promotion to distributed (only a durable is supposed to do that).